src/lib/tcp/src/util/mod.rs

```diff
@@ -0,0 +1,73 @@
+pub mod time;
+
+/// An array that derefs to a variable-length slice. Useful for storing variable-length data in a
+/// struct without allocating. The generic `N` must be a valid `u8`, so can only store up to
+/// `u8::MAX` items.
+#[derive(Copy, Clone, Debug)]
+pub struct SmallArrayBackedSlice<const N: usize, T> {
```

Comment:
Maybe the existing [SmallVec](https://docs.rs/smallvec/latest/smallvec/struct.SmallVec.html) would work for our purposes?

Reply:
I do think we'll need `SmallVec` eventually in some places, but an advantage of `SmallArrayBackedSlice` (which could use a better name) is that it can't grow past the buffer size. For example a TCP packet can have up to 4 selective acknowledgements, so `SmallArrayBackedSlice` allows us to enforce that at the type level, while `SmallVec` has no upper limit on the size.